### PR TITLE
repository for RavenDB + tests to crud operations + initial commit of…

### PR DESCRIPTION
**Pull request by Havret**
… web application

### CODE REVIEW

File: Src/MaintainableSelenium/MaintainableSelenium.API/CategoryRepository.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            {
+                if (!string.IsNullOrEmpty(test.ExpectedImageId))
+                    test.ExpectedImage = Image.FromStream(filesSession.DownloadAsync(test.ExpectedImageId).Result);
+
+                if (!string.IsNullOrEmpty(test.ResultImageId))
+                    test.ExpectedImage = Image.FromStream(filesSession.DownloadAsync(test.ResultImageId).Result);
+            }
+        }
+
+        private void SaveTestImages(IEnumerable<Test> tests, IAsyncFilesSession filesSession)
```
Comment (cezarypiatek) [inline Src/MaintainableSelenium/MaintainableSelenium.API/CategoryRepository.cs]: Why do you pass filesSession as argument to non static method? It's class member.

File: Src/MaintainableSelenium/MaintainableSelenium.API/CategoryRepository.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            {
+                if (!string.IsNullOrEmpty(test.ExpectedImageId))
+                    filesSession.RegisterUpload(test.ExpectedImageId, test.ExpectedImage.ToStream(ImageFormat.Jpeg));
+
+                if (!string.IsNullOrEmpty(test.ResultImageId))
+                    filesSession.RegisterUpload(test.ResultImageId, test.ResultImage.ToStream(ImageFormat.Jpeg));
+            }
+        }
+
+        private void RemoveTestImages(IEnumerable<Test> tests, IAsyncFilesSession filesSession)
```
Comment (cezarypiatek) [inline Src/MaintainableSelenium/MaintainableSelenium.API/CategoryRepository.cs]: Why do you pass filesSession as argument to non static method? It's class member.

File: Src/MaintainableSelenium/MaintainableSelenium.API/CategoryRepository.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        }
+
+        public void Delete(Category entity)
+        {
+            RemoveTestImages(entity.Tests, _asyncFilesSession);
+
+            _documentSession.Delete(entity);
+        }
+
+        private void LoadTestImages(IEnumerable<Test> tests, IAsyncFilesSession filesSession)
```
Comment (cezarypiatek) [inline Src/MaintainableSelenium/MaintainableSelenium.API/CategoryRepository.cs]: Why do you pass filesSession as argument to non static method? It's class member.

File: Src/MaintainableSelenium/MaintainableSelenium.Test/CategoryRepositoryTest.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                Assert.AreEqual(_category.Tests[0].Name, result.Tests[0].Name);
+                Assert.AreEqual(_category.Tests[0].Name, result.Tests[0].Name);
+                Assert.AreEqual(_category.Tests[0].ExpectedImageId, result.Tests[0].ExpectedImageId);
+                Assert.AreEqual(_category.Tests[0].ExpectedImage, result.Tests[0].ExpectedImage);
+                Assert.AreEqual(_category.Tests[0].ResultImageId, result.Tests[0].ResultImageId);
+                Assert.AreEqual(_category.Tests[0].ResultImage, result.Tests[0].ResultImage);
+                Assert.AreEqual(1, _category.Tests.Count);
+
+                target.Delete(result);
+                documentSession.SaveChanges();
```
Comment (cezarypiatek) [inline Src/MaintainableSelenium/MaintainableSelenium.Test/CategoryRepositoryTest.cs]: Please add SaveChanges() method to the repository. It should invoke these two SaveChanges under the hood.

File: Src/MaintainableSelenium/MaintainableSelenium.API/Models/Category.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.Generic;
+
+namespace MaintainableSelenium.API.Models
+{
+    public class Category
+    {
+        public string Id { get; set; }
+        public string Name { get; set; }
+        public Browser Browser { get; set; }
```
Comment (cezarypiatek) [inline Src/MaintainableSelenium/MaintainableSelenium.API/Models/Category.cs]: There could be many versions of the same browser (e.g. IE9, IE10). Maybe you should change field type to string?

**Comment by cezarypiatek** [comment]
Please consider all my suggestions.

**Comment by Havret** [comment]
All suggested changes applied.

**Pull request merged**